File: generated_code/baseline_code_java/code_row_127.java
import java.io.*;

public class http_hello_page {
    public static void main(String[] args) throws Exception{
        FileInputStream request = new FileInputStream("request");
        FileOutputStream response = new FileOutputStream("response"); 
        int c=0;
        String username="";
        while((c=request.read()) != -1){
            if(Character.isLetterOrDigit(c)){
                username += (char)c;
            }
        }
        response.write("Hello ".getBytes());
        response.write(username.getBytes());
    }
}

File: generated_code/baseline_code_c/code_row_127.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void http_hello_page(FILE *request, FILE *response) {
    // Read the request
    char buffer[4096];
    size_t bytes_read = fread(buffer, 1, sizeof(buffer), request);

    // Get the username from the request
    const char *username = NULL;
    for (int i = 0; i < bytes_read; i++) {
        if (buffer[i] == '?') {
            username = buffer + i + strlen("username=");
            break;
        }
    }

    // Construct the response
    char *greeting = NULL;
    if (username != NULL) {
        greeting = malloc(strlen("Hello, ") + strlen(username) + 1);
        sprintf(greeting, "Hello, %s", username);
    } else {
        greeting = "Hello, world";
    }

    // Write the response to the client
    fprintf(response, "HTTP/1.1 200 OK\r\n");
    fprintf(response, "Content-Type: text/plain\r\n");
    fprintf(response, "\r\n");
    fprintf(response, "%s", greeting);
}